dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java

```diff
@@ -83,4 +165,26 @@ public Response getMessages(@Context HttpServletRequest request,
                                 final I18NForm i18NForm) {
         return oldLanguagesResource.getMessages(request, i18NForm);
     }
+
+    private Language saveOrUpdateLanguage(final String languageId, final LanguageForm form) {
+        final Language newLanguage = new Language();
+
+        if (StringUtils.isSet(languageId)) {
+            final Language origLanguage = this.languageAPI.getLanguage(languageId);
+            Sneaky.sneaked(()->BeanUtils.copyProperties(newLanguage, origLanguage));
+        }
+
+        newLanguage.setLanguageCode(form.getLanguageCode());
+        newLanguage.setLanguage(form.getLanguage());
+        newLanguage.setCountryCode(form.getCountryCode());
+        newLanguage.setCountry(form.getCountry());
+
+        this.languageAPI.saveLanguage(newLanguage);
+        return newLanguage;
+    }
+
+    private boolean doesLanguageExist(String languageId) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'languageId' is not assigned and could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4616952)